packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -4497,7 +4487,39 @@ export class AdapterClass extends EventEmitter {
             return tools.maybeCallbackWithError(callback, e);
         }
 
-        this.#objects.findObject(id, type, options || {}, callback);
+        this.#objects.findObject(id, type as ioBroker.CommonType | null, options || {}, callback);
+    }
+
+    /**
+     * Find an object by the exact name or ID.
+     *
+     * @param id exactly object ID (without a namespace)
+     * @param type optional `common.type` of the state: 'number', 'string', 'boolean', 'file', ...
+     * @param options optional user context with language
+     * @param options.user current user
+     * @param options.language language in which the search must be done for multi-language names
+     * @result if the object was found by ID it will return id and may be the multi-language name it exists. If the object was found by name it will return id and the multi-language name. If the object was not found, it will return only name that was searched for.
+     */
+    findForeignObjectAsync(
+        id: string,
+        type: ioBroker.CommonType | null,
+        options: { user?: `system.user.${string}`; language?: ioBroker.Languages },
+    ): Promise<{ id: string | undefined; name: ioBroker.StringOrTranslated | undefined } | null> {
+        return new Promise<{ id: string | undefined; name: ioBroker.StringOrTranslated | undefined } | null>(
```

Comment:
This is formally not matching with the response because both properties exist when id exists, else it is null ... which basically is a breaking change of the response compared to before (because also when ID is missing the name was always returned)